compiler/daml-lf-reader/src/DA/Daml/LF/Reader.hs

```diff
@@ -0,0 +1,43 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.LF.Reader
+    ( Manifest(..)
+    , ManifestData(..)
+    , manifestFromDar
+    ) where
+
+import Codec.Archive.Zip
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.ByteString.Lazy.UTF8 as UTF8
+import qualified Data.HashMap.Strict as Map
+import Data.List.Extra
+import System.FilePath
+
+data Manifest = Manifest { mainDalf :: FilePath , dalfs :: [FilePath] } deriving (Show)
+data ManifestData = ManifestData { mainDalfContent :: BSL.ByteString , dalfsCotent :: [BSL.ByteString] } deriving (Show)
```

Comment:
I'd have split this into separate lines for the fields. Otherwise its hard to write comments.

Reply:
changed